docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+
+```yaml
+spec:
+  params:
+    - name: my-param
+      value: chicken
+```
+
+If the custom task controller knows how to interpret the parameter value, it
+will do so. It might enforce that some parameter values must be specified, or
+reject unknown parameter values.
+
+### Specifying workspaces
+
+If the custom task supports it, you can provide [`Workspaces`](workspaces.md) to share data with the custom task .
+
+```yaml
+spec:
+  workspaces:
+    - name: my-workspace
+      emptyDir: {}
+```
+
+Consult the documentation of the custom task that you are using to determine whether it supports workspaces and how to name them.
+
+### Specifying a ServiceAccount
+
+If the custom task supports it, you can execute the `CustomRun` with a specific set of credentials by
+specifying a `ServiceAccount` object name in the `serviceAccountName` field in your `CustomRun`
+definition. If you do not explicitly specify this, the `CustomRun` executes with the service account
+specified in the `configmap-defaults` `ConfigMap`. If this default is not specified, `CustomRuns`
+will execute with the [`default` service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)
+set for the target [`namespace`](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/).
+
+```yaml
+spec:
+  serviceAccountName: my-account
+```
+
+Consult the documentation of the custom task that you are using to determine whether it supports a service account name.
+
+## Monitoring execution status
+
+As your `CustomRun` executes, its `status` field accumulates information on the
+execution of the `CustomRun`. This information includes start and completion times,
+and any output `results` reported by the custom task controller.
+
+The following example shows the `status` field of a `CustomRun` that has executed
+successfully:
+
+```yaml
+completionTime: "2019-08-12T18:22:57Z"
+conditions:
+  - lastTransitionTime: "2019-08-12T18:22:57Z"
+    message: Execution was successful
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+startTime: "2019-08-12T18:22:51Z"
+```
+
+The following tables shows how to read the overall status of a `CustomRun`:
+
+`status`|Description
+:-------|-----------:
+<unset>|The custom task controller has not taken any action on the Run.
+Unknown|The custom task controller has started execution and the Run is ongoing.
+True|The CustomRun completed successfully.
+False|The CustomRun completed unsuccessfully, and all retries were exhausted.
+
+In any case, the custom task controller should populate the `reason` and
+`message` fields to provide more information about the status of the execution.
+
+### Monitoring `Results`
+
+After the `CustomRun` completes, the custom task controller can report output
+values in the `results` field:
+
+```
+results:
+- name: my-result
+  value: chicken
+```
+
+## Code examples
```

Comment:
@XinruZhang it'd be helpful to implement CustomRun for the Wait Custom Task and reference them in the code examples, but that could be done in another change

Reply:
Agreed, I created an issue to track the example controller implementation: https://github.com/tektoncd/pipeline/issues/5754.

Reply:
sounds great!